lib/Driver/Driver.cpp

```diff
@@ -1102,7 +1118,8 @@ void Driver::buildOutputInfo(const ToolChain &TC, const DerivedArgList &Args,
 
   } else { // DriverKind::Batch
     OI.CompilerMode = OutputInfo::Mode::StandardCompile;
-    if (Args.hasArg(options::OPT_whole_module_optimization))
+    if (Args.hasArg(options::OPT_whole_module_optimization,
+                    options::OPT_index_file))
```

Comment:
This seems wrong. The whole point up till now is that index-file is going to work like single-file mode. Maybe just make jobs non-batchable if they're indexing?

Reply:
I agree it seems wrong. When I tried it the other way, something else complained about the number of output files not matching the number of inputs.
I can look at it again. The other question in my mind was does an indexing job only index the primary file? Or does it index the whole module??

Reply:
Please take a look at Driver::buildInputs in Driver.cpp. The issue is how best to downstream-signal to make the index file path a primary. What I have done is let the argument carry the information, by using the argument with that option in the list. Seems cleanest to me, but it could also be done by adding state somewhere else. Let me know how you feel.

Reply:
I really think the right answer is to have -index-file force single-file mode even if the user passes -wmo, and then -index-file-path controls the one job that gets made.

Reply:
(After talking to @benlangmuir I no longer think this is the case, though it makes me sad because other things are probably wrong because of it. But it's not something that has to change in this PR.)